### DISCUSSION TITLE
v2.0.0-beta03について気づいたことがあれば

### DISCUSSION THREAD

**Discussion by keiji** [cocoa-mhlw/cocoa#865] [General]
v2.0.0-beta03（クローズド）に関するDiscussionです。
iOS版、Android版ともに配信開始済。

 * 接触情報をSecureStorageに保存するようにしました（これまで記録した接触情報は削除されます）。
 * 「基準値未満の接触画面」のスコアを一〇〇分率で表示します。
 * 接触の日時について、UTC基準であることを明確にして、タイムゾーンを変換した期間として表記するようにしました
 * その他、画面デザインを調整しました。

**Comment by kvaluation** [comment]
**■「受信した信号」表示例のスクショ**
日付ごとの情報があって、安心できます。

<p align="left">
  <img src="https://user-images.githubusercontent.com/60703087/154480102-e2a49bc7-c547-4971-ac0f-08b3fc197d6e.png" width="400px)">
</p>


**■ 広範な接触（低リスクすれ違い）**
2/16午後の照合で、13日のすれ違いが表示されました。
<p align="left">
  <img src="https://user-images.githubusercontent.com/60703087/154480495-6a1683f2-e0a1-40a9-972d-51f044dbdbae.png" width="400px)">
</p>

時間の条件を満たさなかったことが明確で、動作検証に有用と思います。
COCOA利用者さまが見た時にも意味が分かりやすく良いと思います。

**■ cocoa-logは、下記です。**
2022/02/17 16:30:26 | Info | DailySummary: 1, but no high-risk exposure detected | ExposureDetectedAsync | /Users/runner/work/1/s/Covid19Radar/Covid19Radar/Services/ExposureDetectionService.cs | 128 | Android | 8.0.0
-- | -- | -- | -- | -- | -- | -- | --

ログの全体
https://docs.google.com/spreadsheets/d/15iCRw2KMUagqdwgYJ5TDNmshJaFGN2meOxL7g_1qhaI/edit#gid=1627779488&range=A881

**Reply by tmurakami** [reply]
「9時」を午後 9時と勘違いそうだな、と思いました。
「午前 9時」とすることはできないでしょうか？
文章が長くなるようならようなら「午前 9時から 24時間」とかどうでしょう？

**Reply by tmurakami** [reply]
もしくは、接触情報の保持期間は 14日なので「2022年」を取ってしまっても良いかもしれません。

**Reply by keiji** [reply]
特に米国では24時間表記ではなくAM/PM表記の方が一般的なので午前/午後表記は必要そうです。
一方「から24時間」も、処理としてはシンプルになって良いですね（年またぎ、月またぎの処理とか書いているので）。

どちらにするか（そもそも変えるかも含めて）、すぐに結論をお伝えできないのがもどかしいところですが、オープンベータもありますし、このあたりは継続的に改善していきたいですね。
方針が見えたらコメントまたはIssueを作る形でお知らせします。

**Reply by keiji** [reply]
#877 を作りました。 午前（AM）を明記します。また、年の表記も年をまたぐときだけ表示するように変更します。

一方、「から 24時間」にすると若干ですが頭の中で計算が必要になるので、なにも考えずに各行を読めばいい現在の表記を維持します。

**Reply by b-wind** [reply]
Locale を考慮するのは分かるのですが、それなら端末の TimeZone は考慮しないのかと言う疑問は出てきますね。
基本的には日本国内で使用するのだからJSTで考えてねと言う事でもおかしくは無いと思いますが。

**Reply by keiji** [reply]
あ、端末のタイムゾーンを考慮して変換しているので、必ずしもJSTになるとは限りません（なので今の注記もJSTという単語を使っていません）が、そういえば利用者以外（医療機関とか）が見ることを考えるとJST固定の方がいいと言う見方もあって、難しいところですね。

**Reply by b-wind** [reply]
> 端末のタイムゾーンを考慮して変換しているので、必ずしもJSTになるとは限りません

そうなんですね。それは失礼しました。素晴らしいと思います。

> 利用者以外（医療機関とか）が見ることを考えるとJST固定の方がいいと言う見方もあって、難しいところですね。

なんとも悩ましい話ですが、医療機関向け表示ページを作るとかもあり得るでしょうか。追加で表示する事も殆ど無さそうなので微妙な所ですが。

**Reply by b-wind** [reply]
時間表現を考慮されるとのことなので、日付の方はどうしたものかなと言う疑問が……（特にUSはこのあたり色々有りすぎる）

https://docs.microsoft.com/ja-jp/dotnet/standard/base-types/standard-date-and-time-format-strings
